### PR TITLE
fix: save the current theme in browser

### PR DESCRIPTION
**Pull request by sanchis**
Persistent current theme.
Fix: #99 

- [x] Set current theme persistent
- [x] Fix styles in darkmode in mission-single view

### CODE REVIEW

**Comment by sanchis** [comment]
Hello, I have some problems to deploy in local.
I got bad gateway (502) when I deploy the server locally.

Can you help me?
image

**Comment by jenaye** [comment]
Hello  @sanchis ! excuse me for delay.
Can you give me the php stacktrace ? `docker-compose logs -f php`

**Comment by sanchis** [comment]
> Hello @sanchis ! excuse me for delay.
> Can you give me the php stacktrace ? `docker-compose logs -f php`

**Logs php** 
```
Package doctrine/reflection is abandoned, you should avoid using it. Use roave/better-reflection instead.
Generating autoload files
composer/package-versions-deprecated: Generating version class...
composer/package-versions-deprecated: ...done generating version class
90 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
Run composer recipes at any time to see the status of your Symfony recipes.
Executing script cache:clear [OK]
Executing script assets:install public [OK]
Waiting for db to be ready...
```

**Logs db**
```
PostgreSQL Database directory appears to contain a database; Skipping initialization
2021-04-13 17:14:39.795 UTC [1] LOG: starting PostgreSQL 12.4 on x86_64-pc-linux-musl, compiled by gcc (Alpine 9.3.0) 9.3.0, 64-bit
2021-04-13 17:14:39.795 UTC [1] LOG: listening on IPv4 address "0.0.0.0", port 5432
2021-04-13 17:14:39.795 UTC [1] LOG: listening on IPv6 address "::", port 5432
2021-04-13 17:14:39.838 UTC [1] LOG: listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2021-04-13 17:14:39.942 UTC [21] LOG: database system was shut down at 2021-04-11 19:35:15 UTC
2021-04-13 17:14:40.008 UTC [1] LOG: database system is ready to accept connections
```
**Logs api**
```
2021/04/13 17:33:18 [error] 6#6: *1 connect() failed (111: Connection refused) while connecting to upstream, client: 172.20.0.1, server: , request: "OPTIONS /authentication_token HTTP/1.1", upstream: "fastcgi://172.20.0.4:9000", host: "localhost:8000", referrer: "http://localhost:4200/"
172.20.0.1 - - [13/Apr/2021:17:33:18 +0000] "OPTIONS /authentication_token HTTP/1.1" 502 565 "http://localhost:4200/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36"
2021/04/13 17:33:27 [error] 6#6: *4 connect() failed (111: Connection refused) while connecting to upstream, client: 172.20.0.1, server: , request: "OPTIONS /authentication_token HTTP/1.1", upstream: "fastcgi://172.20.0.4:9000", host: "localhost:8000", referrer: "http://localhost:4200/"
172.20.0.1 - - [13/Apr/2021:17:33:27 +0000] "OPTIONS /authentication_token HTTP/1.1" 502 565 "http://localhost:4200/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36"
172.20.0.1 - - [13/Apr/2021:17:33:33 +0000] "GET /api HTTP/1.1" 502 565 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36"
2021/04/13 17:33:33 [error] 6#6: *8 connect() failed (111: Connection refused) while connecting to upstream, client: 172.20.0.1, server: , request: "GET /api HTTP/1.1", upstream: "fastcgi://172.20.0.4:9000", host: "localhost:8000"
172.20.0.1 - - [13/Apr/2021:17:33:33 +0000] "GET /favicon.ico HTTP/1.1" 200 32038 "http://localhost:8000/api" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36"
```

File: client/src/app/services/theme.service.ts [hunk 1]
```diff
@@ -9,8 +9,8 @@ export enum Theme {
 
 @Injectable()
 export class ThemeService {
-  private _currentTheme: Theme = Theme.LIGHT_THEME;
-  public readonly onChangeTheme = new BehaviorSubject<Theme>(Theme.LIGHT_THEME);
+  private _currentTheme: Theme = localStorage.getItem('theme') || Theme.LIGHT_THEME;
```
Comment (Darkweak) [inline client/src/app/services/theme.service.ts:12]: Define a new Storage type that extends the AbstractStorage for the theme like the Token storage in the folder `client/src/app/storage`

**Review by Darkweak — CHANGES_REQUESTED** [review]

**Comment by jenaye** [comment]
Thx @sanchis i had the same error, its probably breaking change from last version of containers because the code didnt affect this


The hot-fix for you can be to run only the API  like this ˋdocker-compose up api` this worked for me

**Comment by Darkweak** [comment]
Then could we provide a stable containers stack and upload these on the docker hub to avoid these side effects ?

**Comment by jenaye** [comment]
Hello @sanchis bug fixed,  you can `git pull` then run this `docker-compose up -d --build --force-recreate --remove-orphans`

**Comment by sanchis** [comment]
> Hello @sanchis bug fixed, you can `git pull` then run this `docker-compose up -d --build --force-recreate --remove-orphans`

Sorry but I keep getting the error.
```
172.20.0.1 - - [16/Apr/2021:08:21:08 +0000] "OPTIONS /authentication_token HTTP/1.1" 502 565 "http://localhost:4200/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36"
2021/04/16 08:21:08 [error] 6#6: *1 connect() failed (111: Connection refused) while connecting to upstream, client: 172.20.0.1, server: , request: "OPTIONS /authentication_token HTTP/1.1", upstream: "fastcgi://172.20.0.4:9000", host: "localhost:8000", referrer: "http://localhost:4200/"
2021/04/16 08:22:54 [error] 6#6: *5 connect() failed (111: Connection refused) while connecting to upstream, client: 172.20.0.1, server: , request: "GET /api HTTP/1.1", upstream: "fastcgi://172.20.0.4:9000", host: "localhost:8000"
172.20.0.1 - - [16/Apr/2021:08:22:54 +0000] "GET /api HTTP/1.1" 502 565 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36"
2021/04/16 08:22:55 [error] 6#6: *5 connect() failed (111: Connection refused) while connecting to upstream, client: 172.20.0.1, server: , request: "GET /api HTTP/1.1", upstream: "fastcgi://172.20.0.4:9000", host: "localhost:8000"
172.20.0.1 - - [16/Apr/2021:08:22:55 +0000] "GET /api HTTP/1.1" 502 565 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36"
172.20.0.1 - - [16/Apr/2021:08:22:56 +0000] "GET /favicon.ico HTTP/1.1" 200 32038 "http://localhost:8000/api" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36"
```

**Comment by Darkweak** [comment]
@sanchis 
Did you rebase the branch ?
`git fetch && git rebase --keep-empty origin/master`
`docker pull && docker-compose up -d --build --force-recreate --remove-orphans api`
Then run the client should do the trick

**Comment by sanchis** [comment]
> @sanchis
> Did you rebase the branch ?
> `git fetch && git rebase --keep-empty origin/master`
> `docker pull && docker-compose up -d --build --force-recreate --remove-orphans api`
> Then run the client should do the trick

Yes I get the latest changes in [my fork](https://github.com/sanchis/Smersh/tree/fix/99-theme).

1. `git fetch upstream`
2. `git rebase upstream/master`
3. `git merge origin/mater` (to the current branch `fix/99-theme`)

Then I remove all container related with Smersh and remove all images.
After that I execute `docker-compose up -d --build --force-recreate --remove-orphans api`

**Comment by Darkweak** [comment]
@sanchis did you wait for the `NOTICE: ready to handle connections` when you are starting the container ?
`docker-compose logs -f php` It may take some minutes to be done at the first time

**Comment by sanchis** [comment]
> @sanchis did you wait for the `NOTICE: ready to handle connections` when you are starting the container ?
> `docker-compose logs -f php` It may take some minutes to be done at the first time

**Hello, excuse me for delay.**

Ops I think this is the problem. 😐
phg logs output
```
Package doctrine/reflection is abandoned, you should avoid using it. Use roave/better-reflection instead.
Generating autoload files
composer/package-versions-deprecated: Generating version class...
composer/package-versions-deprecated: ...done generating version class
117 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
Run composer recipes at any time to see the status of your Symfony recipes.
Executing script cache:clear [OK]
Executing script assets:install public [OK]
Waiting for db to be ready...
```

**Comment by Darkweak** [comment]
@sanchis can you run a `make cache` then run `docker-compose down --remove-orphans && make start` ? It seems your php try to access to an unreachable db (possibly due to cache).  
If it doesn't solve your problem can you dump your `docker-compose.yml ` file here ?

**Comment by sanchis** [comment]
> @sanchis can you run a `make cache` then run `docker-compose down --remove-orphans && make start` ? It seems your php try to access to an unreachable db (possibly due to cache).
> If it doesn't solve your problem can you dump your `docker-compose.yml ` file here ?

`make: *** No rule to make target 'start'.  Stop.` I think you mean `make up` ?

**Comment by Darkweak** [comment]
Yes, run `make up` 👍 
Then can you provide us your discord tag to solve your problem on discord pls ?

**Comment by sanchis** [comment]
image
image

Is good for you?
The pull request is ready for review.

Sorry for the problems and delay.

File: client/src/app/storage/Theme.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { AbstractStorage } from "./AbstractStorage";
+
+export class ThemeStorage extends AbstractStorage {
+    protected key = 'theme';
+  }
```
Comment (Darkweak) [inline client/src/app/storage/Theme.ts:5]:
```suggestion
}
```

**Review by Darkweak — APPROVED** [review]
After this last change it looks good to me and could be merged.

File: client/src/app/storage/Theme.ts (hunk 2 above)
Comment (sanchis) [reply client/src/app/storage/Theme.ts]: Fixed in 78e5b06

**Review by jenaye — APPROVED** [review]

**Pull request merged**